### ISSUE TITLE
Crash on startup when using unknown model in lua file

### ISSUE THREAD

**Issue by nallath** [daid/EmptyEpsilon#138]
When trying to add more ships to the shipTemplates.lua, I made a typo in the model to use. This caused a crash on startup.

We really should add better  / more gracefull errror handling, as the scripting should never cause the application to crash.

**Comment by nallath** [comment]
This seems to be caused by this bit of code;
`P<ModelData> ModelData::getModel(string name)
{
    return data_map[name];
}`

When the requested string is unknown, unordered map does the following; 

> If k does not match the key of any element in the container, the function inserts a new element with that key and returns a reference to its mapped value.

I'm not familiar enough with the code to be able to say if we should always do a check or if there are better ways to handle this.

**Comment by daid** [comment]
> We really should add better / more gracefull errror handling, as the scripting should never cause the application to crash.

Currently using an unknown model, or an unknown template causes crashes. No that hard to detect when this is done, but the real question is, how should we handle it then?

**Comment by nallath** [comment]
Show something that is very obviously wrong. A big pink cube would do the trick.

**Comment by Jhovall** [comment]
I disagree, it is pretty obvious in the debug screen, I worked a lot with it (To implement the ships for the larp and it was easy to determine the error).
